livedata/src/main/java/io/getstream/chat/android/livedata/repository/UserRepository.kt

```diff
@@ -5,81 +5,89 @@ import io.getstream.chat.android.client.models.User
 import io.getstream.chat.android.livedata.dao.UserDao
 import io.getstream.chat.android.livedata.entity.UserEntity
 
-class UserRepository(
-    var userDao: UserDao,
-    var cacheSize: Int = 100,
-    var currentUser: User?
+internal class UserRepository(
+    private val userDao: UserDao,
+    private val currentUser: User?,
```

Comment:
Does it make sense to have a "null `currentUser`"?
Can't we delay the `UserRepository` initialization until we have a user logged in?

Reply:
Yes, it's definitely possible, since RepositoryHelper already contains user in its' constructor.